internal/catalogmetadata/sort/sort.go

```diff
@@ -0,0 +1,34 @@
+package sort
+
+import (
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
+)
+
+// ByVersion is a sort "less" function that orders bundles
+// in inverse version order (higher versions on top).
+func ByVersion(b1, b2 *catalogmetadata.Bundle) bool {
```

Comment:
Should we not also sort by package name as we did before with entities? If we don't, what I'm imagining is our list of:
```
- packageName: a-package
  version: 1.0.0
- packageName: b-package
  version: 2.0.0
- packageName: a-package
  version: 3.0.0
```
would get sorted to:
```
- packageName: a-package
  version: 3.0.0
- packageName: b-package
  version: 2.0.0
- packageName: a-package
  version: 1.0.0
```
instead of getting grouped up by name followed by version:
```
- packageName: a-package
  version: 3.0.0
- packageName: a-package
  version: 1.0.0
- packageName: b-package
  version: 2.0.0
```

Reply:
_I think_ ordering by package is not important. We have the following use cases for old bundle entity based `sort.ByChannelAndVersion` at the moment:
* `RequiredPackageVariableSource` - here we always filter by pacakge name so all bundles in sorting will have the same package name
* Twice in `InstalledPackageVariableSource` - on first occasion we call it on a slice filtered by bundle image (so most likely same package) and on second occasion we call it on a slice filtered by `replaces` which should also be within the same package
* `BundlesAndDepsVariableSource` - here is the use case where we might sort different packages. A bundle might require multiple GVKs from different pacakge. But, _I think_, as long as dependencies are sorted by version from newest to oldest we should be ok.

Reply:
I'm not 100% that this is correct, but let's go with it and fix it if it doesn't work. Trying to not overcomplicate it :)